acados/sim/sim_erk_integrator.h

```diff
@@ -25,14 +25,50 @@ extern "C" {
 #endif
 
 #include "acados/sim/sim_common.h"
+#include "acados/utils/external_function.h"
 #include "acados/utils/types.h"
 
+
+
 typedef struct {
+    // Options
+    double interval;
+    int num_stages;
+
+    int num_steps;
+    int num_forw_sens;
+
+    double *A_mat;
+    double *c_vec;
+    double *b_vec;
+
+    bool sens_forw;
+    bool sens_adj;
+    bool sens_hess;
+
+    // Function pointers
+    external_function_fcn_ptrs *forward_vde;
+    external_function_fcn_ptrs *adjoint_vde;
+    external_function_fcn_ptrs *hess_vde;
```

Comment:
Isn't it more logical to have the model of the integrator in sim_in, as it was? I suppose you went for this way to have the nested implementation of submodules but that's not necessary to have (at least initially) for external functions I think. This also adds a lot of casadi-related code in the intergrators atm.

Reply:
On the other hand, it's only possible to change the model and use the same integrator only if you keep exactly the same dimensions, so I am not entirely convinced either what's the best place for the function pointers.

Reply:
Indeed, it is constructed as such to maintain a nested structure of the ``args``, ``memory`` and ``workspace`` structs. Assuming that is the desired approach, it doesn't makes a lot of sense to put the function pointers in the input: Firstly, the in-struct is not an argument to the functions for creating args, memory, and workspace. Secondly, it would suggest to user it is possible to change function pointers on the fly, which isn't the case.

So far the integrator code is completely casadi-agnostic. It is aware that external functions are necessary to compute for instance the VDE sensitivities. The casadi_wrapper is just one (for now the only) tool to instantiate such external functions.